include/kodlab_mjbots_sdk/mjbots_control_loop.h

```diff
@@ -92,33 +112,71 @@ class MjbotsControlLoop : public AbstractRealtimeObject {
    */
   virtual void ProcessInput() {};
 
-  std::shared_ptr<MjbotsRobotInterface>
-      robot_;   /// ptr to the robot object, if unique causes many issues, also should be
-  /// initialized inside thread
+  /*!
+   * @brief Construct a new Setup Options object
+   * 
+   * @param options 
+   */
+  void SetupOptions(const ControlLoopOptions &options);
+
+  std::shared_ptr<RobotClass> robot_;     /// ptr to the robot object
+  std::shared_ptr<kodlab::mjbots::MjbotsRobotInterface> mjbots_interface_;   ///ptr to mjbots_interface object, if unique causes issues, also should be initialized inside thread
```

Comment:
Should this be a unique ptr?

Reply:
Future plans share the robot interface, its possible that it is useful to be able to share the hardware interface as well, and shared pointers are easier to manage. 
Unique pointer would work for now and could just be passed as a raw pointer, but that is probably bad form.